src/includes/class-alma-wc-internationalization.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Alma WooCommerce internationalization class.
+ *
+ * @package Alma_WooCommerce_Gateway
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Not allowed' ); // Exit if accessed directly.
+}
+
+/**
+ * Alma_WC_Internationalization
+ */
+class Alma_WC_Internationalization {
+
+	/**
+	 * Load languages list.
+	 *
+	 * @return array[]
+	 */
+	public static function get_list_languages() {
+
+		if ( self::is_wpml_active() ) {
+			return self::get_wpml_list_languages();
+		}
+
+		return array();
+	}
+
+	/**
+	 * Checks if this website is multilingual.
+	 *
+	 * @return Bool
+	 */
+	public static function is_site_multilingual() {
+		return self::is_wpml_active();
+	}
+
+	/**
+	 * Checks if WPML is active.
+	 *
+	 * @return Bool
+	 */
+	public static function is_wpml_active() {
+		return (bool) did_action( 'wpml_loaded' );
+	}
+
+	/**
+	 * Load WPML languages list.
+	 *
+	 * @return array[]
+	 */
+	public static function get_wpml_list_languages() {
+		$languages_list = array();
+		foreach ( icl_get_languages() as $infos_lang ) {
+			$languages_list[ self::map_locale( $infos_lang['default_locale'] ) ] = $infos_lang['translated_name'];
+		}
+		return $languages_list;
+	}
+
+	/**
+	 * Returns locale in format xx_XX for WP locale format compliance.
+	 * The value returned by this function is the key language code for the <select> of languages.
+	 * This value MUST be the same than the one returned by the get_locale() function.
+	 *
+	 * @param string $locale The locale.
+	 *
+	 * @return string $locale The locale formatted on WordPress way.
+	 */
+	public static function map_locale( $locale ) {
+		$locale = str_replace( '-', '_', $locale );
+		if ( 2 === strlen( $locale ) ) {
+			$locale .= '_' . strtoupper( $locale );
+		}
+		return $locale;
+	}
+
+	/**
+	 * Returns a field translated.
+	 *
+	 * @param string $field The field to translate.
+	 *
+	 * @return mixed
+	 */
+	public static function get_field( $field ) {
+		$field_translated = alma_wc_plugin()->settings->default_settings()[ $field ];
+
+		if ( alma_wc_plugin()->settings->{$field . '_' . get_locale() } && self::is_site_multilingual() ) {
+			$field_translated = alma_wc_plugin()->settings->{$field . '_' . get_locale() };
+		} elseif ( alma_wc_plugin()->settings->{$field} ) {
+			$field_translated = alma_wc_plugin()->settings->{$field};
+		}
+
+		return $field_translated;
+	}
+
+	/**
+	 * Get string translated in another language.
+	 *
+	 * @param string $string String to translate, in default language (english).
+	 * @param string $code_lang The desired language code for the translated string.
+	 *
+	 * @return string
+	 */
+	public static function get_translated_text( $string, $code_lang ) {
+		$translation = $string;
+
+		$mo_file = ALMA_WC_PLUGIN_PATH . 'languages/alma-woocommerce-gateway-' . $code_lang . '.mo';
+		if ( ! file_exists( $mo_file ) ) {
+			return $translation;
+		}
+
+		$my_mo = new MO();
+		$my_mo->import_from_file( $mo_file );
```

Comment:
here, you will loop over 3 fields & 5 langs at least each time you are loading the admin page.
So you will read 15 files each time.

to avoid this high number of reads... you can store $mo in private static property (`$mo_resources` for example ) ...

```
if (array_key_exists($code_lang, self::$mo_resources)) {
  return self::translate_with( $mo_resources($code_lang), $translation );
}
if ( ! file_exists( $mo_file ) ) {
 return $translation;
}
$mo = new MO();
$mo->import_from_file( $mo_file );
$mo_resources[$code_lang] = $mo;
return self::translate_with( $mo, $translation );
```